### ISSUE TITLE
Feature Request:  HTTP component needs some way to cancel it.

### ISSUE THREAD

**Issue by NathanaelA** [NativeScript/NativeScript#1845]
### Tell us about the problem
HTTP Component has no way to cancel it.   The internal timeout only works in some situations; and currently I have a server that is responding improperly (i.e. the socket is opened but the socket goes into an await state; and no data comes back, and so the http timeout doesn't think anything is wrong)

### Which platform(s) does your issue occur on?
Android (Haven't checked iOS)

### Please provide the following version numbers that your issue occurs with:
1.7.1 

### Please tell us how to recreate the issue in as much detail as possible.
No easy way to duplicate it.  You have to have a server which gets into a state where the port is opened in response; but no data is transmitted.   So if I had some way to "cancel" the transfer; or some way that if the timeout occurs with NO data received that it would cancel.

### Is there code involved? 
No

**Comment by enchev** [comment]
Hey @NathanaelA,

Considering the fact that [our API](https://github.com/NativeScript/NativeScript/blob/master/http/http.d.ts) is using promises right now do you have any idea how to implemented this feature?

**Comment by NathanaelA** [comment]
Yeah, we can do the exact same thing that I did with Animations.  Add "cancel" to the promise.  

Maybe we should abstract it a little and make a base "cancelablePromise" and have the Animation API and this one (and I'm sure others in the future) use a cancelable Promise.  Seems like something that would be very useful.

**Comment by dreambo8563** [comment]
it sounds like unsubscribe the stream in Rxjs?

**Comment by manojdcoder** [comment]
Any update on this feature? Any chance for including this in any near future release?

**Comment by 3rror404** [comment]
@NathanaelA @enchev I could really use this feature. Is this likely to happen?

**Comment by ickata** [comment]
+1

**Comment by morzyns** [comment]
+1

**Comment by seveneye** [comment]
+1

**Comment by rafarx** [comment]
+1

**Comment by KonstantinObuhov** [comment]
+1

**Comment by Lehren** [comment]
+1

**Comment by jnorkus** [comment]
+1

**Comment by agonper** [comment]
+1
